runtime/query.go

```diff
@@ -78,21 +76,18 @@ func (r *Runtime) Query(ctx context.Context, instanceID string, query Query, pri
 			continue
 		}
 
-		// Infer OLAP connector for common resource types used in a query
-		if mv := res.GetMetricsView(); mv != nil {
-			if mv.State.ValidSpec == nil {
-				return fmt.Errorf("metrics view spec is not valid")
-			}
-			olapConnector = mv.State.ValidSpec.Connector
-		} else if mdl := res.GetModel(); mdl != nil {
-			olapConnector = mdl.Spec.OutputConnector
-		} else if src := res.GetSource(); src != nil {
-			olapConnector = src.Spec.SinkConnector
-		}
-
 		// Add to cache key.
 		// Using StateUpdatedOn instead of StateVersion because the state version is reset when the resource is deleted and recreated.
 		key := fmt.Sprintf("%s:%s:%d:%d", res.Meta.Name.Kind, res.Meta.Name.Name, res.Meta.StateUpdatedOn.Seconds, res.Meta.StateUpdatedOn.Nanos/int32(time.Millisecond))
+		if mv := res.GetMetricsView(); mv != nil {
+			cacheKey, err := r.metricsViewCacheKey(ctx, instanceID, res.Meta.Name.Name, priority)
+			if err != nil {
+				// skip caching
+				// the cache_key_resolver should ideally only return an error if caching is disabled or context is cancelled
+				return query.Resolve(ctx, r, instanceID, priority)
+			}
```

Comment:
Can we have a dedicated `skip bool` for skipping caching. Because if the `err` is a context cancellation, that's ideally an error we return immediately. It will also make it easier to understand if an unexpected error is actually returned.

Reply:
Okay. In that case I think it will be best to expose `errCachingDisabled` and check for it directly for now. Once query package is removed that can be made internal as well.